src/metax_api/models/catalog_record_v2.py

```diff
@@ -0,0 +1,1065 @@
+# This file is part of the Metax API service
+#
+# Copyright 2017-2018 Ministry of Education and Culture, Finland
+#
+# :author: CSC - IT Center for Science Ltd., Espoo Finland <servicedesk@csc.fi>
+# :license: MIT
+
+from collections import defaultdict
+from copy import deepcopy
+import logging
+
+from django.conf import settings
+from django.db.models import Q
+
+from metax_api.exceptions import Http400, Http403
+from metax_api.utils import (
+    datetime_to_str,
+    DelayedLog,
+    generate_doi_identifier,
+    generate_uuid_identifier,
+    get_identifier_type,
+    get_tz_aware_now_without_micros,
+    IdentifierType,
+)
+from .common import Common
+from .directory import Directory
+from .file import File
+from .catalog_record import (
+    CatalogRecord,
+    DataciteDOIUpdate,
+    DatasetVersionSet,
+    RabbitMQPublishRecord,
+    REMSUpdate,
+)
+
+
+_logger = logging.getLogger(__name__)
+
+
+"""
+A version V2 proxy model for CatalogRecord.
+
+Note that in this class, calling super().some_method() actually calls CatalogRecord.some_method(),
+instead of CommonModel.some_method() !!! Therefore, if wishing to call the actual CommonModel method,
+or even the base Model, super(Common, self).some_method() should be called instead. Calling just
+super().some_method() can cause some unnecessary processing and prevents editing some fields, so
+may want to avoid that.
+
+At least UNTIL this class fully replaces the V1 version.
+"""
+
+
+class CatalogRecordV2(CatalogRecord):
+
+    class Meta:
+        # CatalogRecordV2 operates on the same database table as CatalogRecord model. Only the class
+        # behaviour may differ from base class.
+        proxy = True
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        from metax_api.api.rest.v2.serializers import CatalogRecordSerializerV2
+        self.serializer_class = CatalogRecordSerializerV2
+
+    def save(self, *args, **kwargs):
+        if self._operation_is_create():
+            self._pre_create_operations()
+            super(CatalogRecord, self).save(*args, **kwargs)
+            self._post_create_operations(pid_type=kwargs.pop('pid_type', None))
+        else:
+            self._pre_update_operations()
+            super(CatalogRecord, self).save(*args, **kwargs)
+            self._post_update_operations()
```

Comment:
"PUT /rest/v2/datasets/{identifier}/files/user_metadata" returns  HTTP 200 and works fine but doesn't print anything in return... I don't know if it should? For example /rest/v2/datasets/{identifier} does...